src/sieve.py

```diff
@@ -15,6 +16,18 @@ def sieve(n: int) -> list[int]:
     candidates = list(range(2, n + 1))
     primes = []
 
+    while candidates:
+        p=candidates[0]
+        #print(p)
+        for i in candidates:
+            if i%p == 0:
+                candidates.remove(i)
```

Comment:
For reasons I haven't explained (yet) it is an expensive operation to remove elements from a list (unless it is the last element). Instead of removing from `candidates` you are actually better off building a whole new list of the remaining candidates. Something like this will do the trick:

```python
candidates = [cand for cand in candidates if cand % p != 0]
```